api_tests/preprints/views/test_preprint_versions.py

```diff
@@ -154,3 +155,377 @@ def test_permission_error(self):
         res = self.app.post_json_api(self.post_mod_version_create_url, auth=user_write.auth, expect_errors=True)
         assert res.status_code == 403
+
+
+class TestPreprintVersionsListRetrieve(ApiTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.user = AuthUserFactory()
+        self.admin = AuthUserFactory()
+        self.moderator = AuthUserFactory()
+        self.provider = PreprintProviderFactory(_id='osf', name='osfprovider')
+        self.post_mod_preprint = PreprintFactory(
+            reviews_workflow='post-moderation',
+            is_published=True,
+            creator=self.user,
+            provider=self.provider
+        )
+        self.post_mod_preprint.add_contributor(self.admin, permissions=permissions.ADMIN)
+        self.post_mod_preprint.add_contributor(self.user, permissions=permissions.WRITE)
+        self.post_mod_preprint.add_contributor(self.user, permissions=permissions.READ)
+
+        self.pre_mod_preprint = PreprintFactory(
+            reviews_workflow='pre-moderation',
+            is_published=False,
+            creator=self.user,
+            provider=self.provider
+        )
+        self.pre_mod_preprint.add_contributor(self.admin, permissions=permissions.ADMIN)
+        self.pre_mod_preprint.add_contributor(self.user, permissions=permissions.WRITE)
+        self.pre_mod_preprint.add_contributor(self.user, permissions=permissions.READ)
+
+        self.pre_mod_preprint.provider.get_group('moderator').user_set.add(self.moderator)
+        self.post_mod_preprint.provider.get_group('moderator').user_set.add(self.moderator)
+        self.latest_preprint = self.post_mod_preprint
+        self.origin_guid = str(self.post_mod_preprint.guids.first()._id)
+        for _ in range(5):
+            new_version = PreprintFactory.create_version(
+                create_from=self.latest_preprint,
+                creator=self.user,
+                set_doi=False
+            )
+            self.latest_preprint = new_version
+            new_version.add_contributor(self.admin, permissions=permissions.ADMIN)
+            new_version.add_contributor(self.user, permissions=permissions.WRITE)
+            new_version.add_contributor(self.user, permissions=permissions.READ)
+
+        self.version_list_url = f"/{API_BASE}preprints/{self.origin_guid}_v1/versions/"
+
+    def test_list_versions(self):
+        res = self.app.get(self.version_list_url, auth=self.user.auth)
+        assert res.status_code == 200
+        data = res.json['data']
+        assert len(data) == 6
+        assert len(set([item['id'] for item in data])) == 6
+
+    def test_filter_by_id(self):
+        version = self.latest_preprint.versioned_guids.first()
+        res = self.app.get(f"{self.version_list_url}?filter[id]={version.referent._id}", auth=self.user.auth)
+        assert res.status_code == 200
+        data = res.json['data']
+        assert len(data) == 6
+        assert data[0]['id'] == version.referent._id
```

Comment:
Remove the filter tests. We don't use filter for this endpoint. And the way filtering by `id` doesn't make sense.